src/util/utilFunctions.js

```diff
@@ -0,0 +1,36 @@
+import { AUTO, autoTimeIntervals } from '../constants'
+
+/** getWindowAndResolution
+ *
+ * @param autoWinResConfig is an array of Object with length always greater than 1,
+ * i.e config array should contain at least 1 object
+ * @param timeRange
+ * @returns {{window, resolution}}
+ */
+const getWindowAndResolution = (autoWinResConfig, timeRange) => {
+  let i = 0
+  do { i++ } while (i < autoWinResConfig.length - 1 && timeRange >= autoWinResConfig[i][0])
+  const { window, resolution } = autoWinResConfig[i][1]
+  return { window, resolution }
+}
+
+/** resolveAutoWinRes
+ *
+ * @param windowSelected
+ * @param resolutionSelected
+ * @param timeRangeSelected
+ * @returns {{window: *, resolution: *}}
+ */
+export const resolveAutoWinRes = (windowSelected, resolutionSelected, timeRangeSelected) => {
+  let { window, resolution } = getWindowAndResolution(autoTimeIntervals, timeRangeSelected)
```

Comment:
Minor comments
1. it will always run the getWindowAndResolution function.Even if it's not necessary.
2.if window and resolution are not redefined it better uses `const` instead of `let`


const { window, resolution } = getWindowAndResolution(autoTimeIntervals, timeRangeSelected);

Reply:
> Minor comments
> 
> 1. it will always run the getWindowAndResolution function.Even if it's not necessary.
Done! Optimized. Added to the next commit
> 2.if window and resolution are not redefined it better uses `const` instead of `let`.
Done ! Optimized. Added to the next commit 
> 3. I guess we can keep one parameter - timeRandeSelected, because autoTimeIntervals is constant.
> I am not sure about this. Coz we already have a parameter. Also, I have replace the milliseconds logic with days logic. So, please take a look again.
> const { window, resolution } = getWindowAndResolution(autoTimeIntervals, timeRangeSelected);